src/devices/fakeDepthCamera/fakeDepthCameraDriver.cpp

```diff
@@ -0,0 +1,248 @@
+    image(YARP_NULLPTR)
+{}
+
+fakeDepthCameraDriver::~fakeDepthCameraDriver()
+{
+}
+
+bool fakeDepthCameraDriver::open(Searchable& config)
+{
+    Property cfg;
+    cfg.put("device", "test_grabber");
+    testgrabber.open(cfg);
+    testgrabber.view(image);
+
+    vector<tuple<double*, string, double> > param;
+    param.push_back(make_tuple(&rgb_h,    "rgb_h",    480.0));
+    param.push_back(make_tuple(&rgb_w,    "rgb_w",    640.0));
+    param.push_back(make_tuple(&dep_h,    "rgb_h",    480.0));
+    param.push_back(make_tuple(&dep_w,    "rgb_w",    640.0));
+    param.push_back(make_tuple(&accuracy, "accuracy", 0.001));
+    param.push_back(make_tuple(&rgb_Vfov, "rgb_Vfov", 50.0));
+    param.push_back(make_tuple(&rgb_Hfov, "rgb_Hfov", 36.0));
+    param.push_back(make_tuple(&dep_Vfov, "dep_Vfov", 50.0));
+    param.push_back(make_tuple(&dep_Hfov, "dep_Hfov", 36.0));
+    param.push_back(make_tuple(&dep_near, "dep_near", 0.2));
+    param.push_back(make_tuple(&dep_far,  "dep_far",  6.0));
+    for (auto p : param)
+    {
+        if (config.check(get<1>(p)))
+        {
+            *get<0>(p) = config.find(get<1>(p)).asDouble();
+        }
+        else
+        {
+            *get<0>(p) = get<2>(p);
+        }
+        
+    }
+    
+    return true;
+}
+
+bool fakeDepthCameraDriver::close()
+{
+    return true;
+}
+
+int fakeDepthCameraDriver::getRgbHeight()
+{
+    return image->height();
+}
+
+int fakeDepthCameraDriver::getRgbWidth()
+{
+    return image->width();
+}
+
+bool  fakeDepthCameraDriver::getRgbSupportedConfigurations(yarp::sig::VectorOf<CameraConfig> &configurations){
+    yWarning()<<"fakeDepthCameraDriver:getRgbSupportedConfigurations not implemented yet";
+    return false;
+}
+bool fakeDepthCameraDriver::getRgbResolution(int &width, int &height)
+{
+    width  = image->width();
+    height = image->height();
+    return true;
+}
+
+bool yarp::dev::fakeDepthCameraDriver::setRgbResolution(int width, int height)
+{
+    return false;
+}
+
+bool fakeDepthCameraDriver::setDepthResolution(int width, int height)
+{
+    return false;
+}
+
+bool fakeDepthCameraDriver::setRgbFOV(double horizontalFov, double verticalFov)
+{
+    rgb_Hfov = horizontalFov;
+    rgb_Vfov = verticalFov;
+    return true;
+}
+
+bool fakeDepthCameraDriver::setDepthFOV(double horizontalFov, double verticalFov)
+{
+    dep_Hfov = horizontalFov;
+    dep_Vfov = verticalFov;
+    return true;
+}
+
+bool fakeDepthCameraDriver::setDepthAccuracy(double in_accuracy)
+{
+    accuracy = in_accuracy;
+    return true;
+}
+
+bool fakeDepthCameraDriver::getRgbFOV(double &horizontalFov, double &verticalFov)
+{
+    horizontalFov = rgb_Hfov;
+    verticalFov   = rgb_Vfov;
+    return false;
+}
+
+bool fakeDepthCameraDriver::getRgbMirroring(bool& mirror)
+{
+    return false;
+}
+
+bool fakeDepthCameraDriver::setRgbMirroring(bool mirror)
+{
+    return false;
+}
+
+bool fakeDepthCameraDriver::getRgbIntrinsicParam(Property& intrinsic)
+{
+    return false;
```

Comment:
Can we set some 'identity element' as return value?
The reason is that if a program requires those values, it can go forward instead of quitting early due to missing info